### PR TITLE
[TASK] Mention type casting for FlexForms

### PR DESCRIPTION
**Pull request by DanielSiepmann**

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
Please check syntax, etc. I didn't write this kind of rst for a while …

File: Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst [hunk 1]
```diff
@@ -126,6 +126,11 @@ must be arrays.)
    :Child elements:
          <[field name]>
 
+Elements can use the attribute type to define their type, e.g. explicitly use boolean.
+An example would look like:
+
+.. code-block: xml
+   <required type="boolean">1<required>
```
Comment (DanielSiepmann) [inline Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst:133]:
```suggestion
   <required type="boolean">1</required>
```

File: Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst [hunk 2]
```diff
@@ -126,6 +126,11 @@ must be arrays.)
    :Child elements:
          <[field name]>
 
+Elements can use the attribute type to define their type, e.g. explicitly use boolean.
+An example would look like:
+
+.. code-block: xml
+   <required type="boolean">1</required>
```
Comment (linawolf) [inline Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst:133]:
```suggestion
..  code-block: xml

    <required type="boolean">1</required>
```

**Comment by linawolf** [comment]
@DanielSiepmann should this be backported to TYPO3 v12?

**Comment by DanielSiepmann** [comment]
Sure, please. Also v11.5 and probably even older versions have this feature:
- main: https://github.com/TYPO3/typo3/blob/main/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L1354
- v12: https://github.com/TYPO3/typo3/blob/12.4/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L1488
- v11: https://github.com/TYPO3/typo3/blob/11.5/typo3/sysext/core/Classes/Utility/GeneralUtility.php#L1589

File: Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst [hunk 3]
```diff
@@ -126,6 +126,12 @@ must be arrays.)
    :Child elements:
          <[field name]>
 
+Elements can use the attribute type to define their type, for example explicitly use boolean.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/DataFormats/T3datastructure/Elements/Index.rst:129]:
```suggestion
Elements can use the attribute :xml:`type` to define their type, for example explicitly use boolean.
```

**Review by brotkrueml — APPROVED** [review]

**Comment by brotkrueml** [comment]
@DanielSiepmann Thank you for the improvement :+1:

**Pull request merged**